pycode/memilio-epidata/memilio/epidata/modifyDataframeSeries.py

```diff
@@ -209,8 +210,16 @@ def extract_subframe_based_on_dates(df, start_date, end_date):
     @param df The dataframe which has to be edited
     @param start_date Date of first date in dataframe
     @param end_date Date of last date in dataframe
+    @param moving_average start_date and end_date are extended by half of moving_average to ensure an accurate calculation.
     """
 
+    # shift start and end date for relevant dates to compute moving average.
+    # if moving average is odd, both dates are shifted equaly.
+    # if moving average is even, start date is shifted one day more than end date.
+    if moving_average > 0:
+        end_date = end_date + timedelta(int(np.floor((moving_average-1)/2)))
+        start_date = start_date - timedelta(int(np.ceil((moving_average-1)/2)))
```

Comment:
I do not like this handling. Yes, if a moving average should be computed, it'd be good to use the data left of start and right of end before computing. However, we should only return what the user asked for.

What we could do is calling the moving average function if ma>0 before cutting. I think that would be a better option.

Reply:
Changed it.